lib/internal/module.js

```diff
@@ -3,11 +3,34 @@
 exports = module.exports = {
   makeRequireFunction,
   stripBOM,
-  addBuiltinLibsToObject
+  addBuiltinLibsToObject,
+  applyTransforms
 };
 
 exports.requireDepth = 0;
 
+const transforms = [];
+
+function addTransform(transform) {
+  transforms.push(transform)
+}
+function removeTransform(transform) {
+  const index = transforms.indexOf(transform)
+  if (index >= 0) {
+    transforms.splice(index, 1)
+  }
+}
+
+function applyTransforms(content, filename) {
+  return transforms.reduce((content, transform) => {
+    const next = transform(content, filename);
+    if (typeof next !== 'string') {
+      throw new Error('Module transforms must return the modified content')
+    }
+    return next;
+  }, content);
+}
+
```

Comment:
[premature optimization is the root of all evil](http://wiki.c2.com/?PrematureOptimization)...
IMHO code duplication is worse than a minor performance impact. I'm sure the actual `compile` is orders of magnitude heavier.
Also EE are super fast when there are no listeners.
Let's test and see...